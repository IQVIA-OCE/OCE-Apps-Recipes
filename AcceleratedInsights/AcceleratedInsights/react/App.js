import React, { useState, useEffect } from "react";
import { StyleSheet, Linking, Platform, View, Text, useColorScheme } from "react-native";
import { WebView } from "react-native-webview";
import { Button, DarkTheme, DefaultTheme, Loader, Provider as ApolloProvider} from "@oce-apps/apollo-react-native";
import Icon from "react-native-vector-icons/MaterialCommunityIcons";
import { layoutBridge } from "@oce-apps/oce-apps-bridges";
import { queryWithSOQL } from './src/helpers';

Icon.loadFont();

const App = ({ recordId, isTest = false }) => {
  const [uri, setUri] = useState("");
  const [dataFetched, setDataFetched] = useState(false);

  const webview = null;

  let preferredTheme = DefaultTheme;
  let url1 = "";

  const colorScheme = useColorScheme();

  if (Platform.OS !== 'web') {
      preferredTheme = colorScheme === "dark" ? DarkTheme : DefaultTheme;
  }
  else
  {
    preferredTheme = DefaultTheme;
  }

  const handleWebViewNavigationStateChange = (newNavState) => {
    const { url } = newNavState;

    if (!url) return;

    if (url !== uri) {
      this.webview.stopLoading();
      this.webview.goBack();
      Linking.openURL(url);
    }
  };

  const testFuncsOverride2 = () => {
    if (url1 == "https://google.com")
      return true;
    else
      return false;
  };

  const testFuncsOverride3 = () => {
    if (url2 == "https://google.com")
      return true;
    else
      return false;
  };

  const testFuncsOverride = () => {
    testFuncsOverride2();
  };

  const fetchAccountData = async () => {
    if (recordId)
    {
        let query = `select Id, QIDC__OrchestrationLoginTokenLong__c from Account where Id = '${recordId}'`;
        let records = await queryWithSOQL(query);
        if (records.length > 0) {
        setUri(records[0].QIDC__OrchestrationLoginTokenLong__c);
        setDataFetched(true);
        }
    }
  };  

  useEffect(() => {
    fetchAccountData();
  }, []);

  const testFuncs = () => {
    this.webview = WebView;
    handleWebViewNavigationStateChange({ url: "" });
    url1 = "https://google.com";
    url2 = "https://google.com";
    //handleWebViewNavigationStateChange({ url: "https://google.com" });
    testFuncsOverride();
    testFuncsOverride3();
    url1 = "https://google1.com";
    url2 = "https://google1.com";
    testFuncsOverride();
    testFuncsOverride3();
  };

  if (uri)
  {
    if (uri.length > 0 && dataFetched)
    {
      if (Platform.OS === 'web') {
        layoutBridge.setHeight(80);
        return (
          <View style={styles.container}>
            <Text testID="web-test" style={styles.foundText}
            onPress={() => Linking.openURL(uri)}>
              Click here to access to Accelerated Insights.
            </Text>
          </View>
        );
      }
      layoutBridge.setHeight(600);  
      return (
          <ApolloProvider theme={preferredTheme}>
          <WebView
            ref={(ref) => (this.webview = ref)}
            allowsBackForwardNavigationGestures={true}
            source={{ uri: uri }}
            onNavigationStateChange={this.handleWebViewNavigationStateChange}
            testID="webview-test"
            renderLoading={()=>{return(<Loader />)}}
            startInLoadingState={true}
          />
          { isTest &&
            <View>
              <Button
                testID='buttonFuncs-test'
                title="buttonFuncs-test"
                onPress={() => testFuncs()}
              />
            </View>
          }
        </ApolloProvider>
      );
    }
  }
  layoutBridge.setHeight(80);
  return (
    <ApolloProvider theme={preferredTheme}>
      <View style={styles.container}>
        <Text style={styles.notFoundText}>Accelerated Insights not found for this account.</Text>
      </View>
      { isTest &&
        <View>
          <Button
            testID='buttonFuncs-test'
            title="buttonFuncs-test"
            onPress={() => testFuncs()}
          />
        </View>
      }
    </ApolloProvider>
  );
};

const styles = StyleSheet.create({
  container: {
    justifyContent: "center",
    alignItems: "center"
  },
  notFoundText: {
    paddingHorizontal: 15,
    paddingVertical: 25,
    fontWeight: 'bold',
    fontSize: 20,
    color:'red',
  },
  foundText:
  {
    paddingHorizontal: 15,
    paddingVertical: 25,
    fontWeight: 'bold',
    fontSize: 20,
    color:'blue',
  }

});

export default App;

