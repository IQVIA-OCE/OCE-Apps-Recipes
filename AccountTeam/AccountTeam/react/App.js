import React from 'react';
import { NativeModules, ScrollView, Platform } from 'react-native';
import { sfNetAPI } from 'oce-apps-bridges';
import { Provider as ReduxProvider } from 'react-redux';
import { Provider as ApolloProvider } from 'apollo-react-native';
import { AccountTeam } from './src/screens/AccountTeam';
import { store } from './src/store/store';

export default App = ({ instanceId, recordId }) => {

    sfNetAPI.enablePromises();
    if (instanceId) {
        const layout = NativeModules.LayoutBridge;
        layout.setHeight(600, instanceId);
    }
    if (Platform.OS === 'web') {
        return (
            <ReduxProvider store={store}>
                <ApolloProvider>
                    <ScrollView>
                        <AccountTeam accountId={recordId} />
                    </ScrollView>
                </ApolloProvider>
            </ReduxProvider >
        );
    }
    return (
        <ReduxProvider store={store}>
            <ApolloProvider>
                <AccountTeam accountId={recordId} />
            </ApolloProvider>
        </ReduxProvider >
    );
};
