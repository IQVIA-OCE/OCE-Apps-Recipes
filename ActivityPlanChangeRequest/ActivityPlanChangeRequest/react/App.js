import React, { useEffect } from 'react';
import { Provider, useDispatch } from 'react-redux';
import { Portal, Provider as ApolloProvider } from 'apollo-react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import store from './src/store/store';
import UpdateActivityPlan from './src/screens/UpdateActivityPlan/UpdateActivityPlan';
import {
  getAccountInfo,
  getDescribedAPCR,
  getAccountsList,
  getPlanCycleDates,
} from './src/store/applicationSlice/applicationSlice';
import NewActivityPlanChangeRequest from './src/screens/NewActivityPlanChangeRequest/NewActivityPlanChangeRequest';
import RemoveActivityPlan from './src/screens/RemoveActivityPlan/RemoveActivityPlan';
import { createStackNavigator } from 'react-navigation-stack';
import { NEW_ACTIVITY } from './src/constants/routes';
import { createAppContainer } from 'react-navigation';
import NotificationBanner from './src/components/NotificationBanner/NotificationBanner';
import Loader from './src/components/Loader/Loader';
import { NativeModules } from 'react-native';
import AddActivityToPlan from './src/screens/AddActivityToPlan/AddActivityToPlan';

Icon.loadFont();

const MainNavigator = createStackNavigator(
  {
    NewActivityPlanChangeRequest: {
      screen: NewActivityPlanChangeRequest,
      navigationOptions: {
        header: null,
      },
    },
    UpdateActivityPlan: {
      screen: UpdateActivityPlan,
      navigationOptions: {
        header: null,
      },
    },
    RemoveActivityPlan: {
      screen: RemoveActivityPlan,
      navigationOptions: {
        header: null,
      },
    },
    AddActivityToPlan: {
      screen: AddActivityToPlan,
      navigationOptions: {
        header: null,
      },
    },
  },
  {
    initialRouteName: NEW_ACTIVITY,
  }
);
const Navigation = createAppContainer(MainNavigator);

const Root = ({ parentId }) => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getDescribedAPCR());
    dispatch(getAccountInfo(parentId));
    dispatch(getAccountsList());
    dispatch(getPlanCycleDates(parentId));
  }, []);
  return (
    <ApolloProvider>
      <Portal>
        <Loader />
        <NotificationBanner />
      </Portal>
      <Navigation />
    </ApolloProvider>
  );
};

const App = ({ recordId, instanceId }) => {
  if (instanceId) {
    const layout = NativeModules.LayoutBridge;
    layout.setHeight(500, instanceId);
  }

  return (
    <Provider store={store}>
      <Root parentId={recordId} testID="root"/>
    </Provider>
  )
}

export default App;
