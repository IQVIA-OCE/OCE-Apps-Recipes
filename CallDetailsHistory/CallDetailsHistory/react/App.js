import React from 'react';
import { Platform, ScrollView } from 'react-native';
import CallDetailsHistory from './src/CallDetailsHistory';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import { Provider } from 'apollo-react-native';
import { layoutBridge } from 'oce-apps-bridges';

Icon.loadFont();

export const isIphone = !Platform.isPad;

export const NAMESPACE = 'OCE__';

export default props => {
  const { instanceId, recordId } = props;

  if (instanceId) {
    layoutBridge.setHeight(600);
  }

  if (isIphone) {
    return (
      <Provider>
        <ScrollView>
          <CallDetailsHistory recordId={recordId} />
        </ScrollView>
      </Provider>
    )
  } else {
    return (
      <Provider>
        <CallDetailsHistory recordId={recordId} />
      </Provider>
    )
  }
}
