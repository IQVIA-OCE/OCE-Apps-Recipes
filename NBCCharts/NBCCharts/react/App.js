import { Platform, useColorScheme } from "react-native";
import { createAppContainer } from "react-navigation";
import { createStackNavigator } from "react-navigation-stack";
import { DarkTheme, DefaultTheme, Provider } from "apollo-react-native";
import AccountsScreen from "./src/screens/accounts";
import AccountDetailsScreen from "./src/screens/accountDetails";
import AccountInfoScreen from "./src/screens/accountInfo";
import AppBar from "./src/components/presentation/AppBar";
import React from "react";
import Icon from "react-native-vector-icons/MaterialCommunityIcons";

Icon.loadFont();

const MainNavigator = createStackNavigator({
  AccountsList: {
    screen: AccountsScreen,
    navigationOptions: ({ navigation, navigationOptions }) => ({
      header: props => (
        <AppBar
          {...props}
          navigation={navigation}
          navigationOptions={navigationOptions}
        />
      ),
      headerStyle: {
        backgroundColor: "transparent"
      }
    })
  },
  AccountDetails: {
    screen: AccountDetailsScreen,
    navigationOptions: ({ navigation, navigationOptions }) => ({
      header: props => (
        <AppBar
          {...props}
          navigation={navigation}
          navigationOptions={navigationOptions}
        />
      ),
      headerStyle: {
        backgroundColor: "transparent"
      }
    })
  },
  AccountInfo: {
    screen: AccountInfoScreen,
    navigationOptions: ({ navigation, navigationOptions }) => ({
      header: props => (
        <AppBar
          {...props}
          navigation={navigation}
          navigationOptions={navigationOptions}
        />
      ),
      headerStyle: {
        backgroundColor: "transparent"
      }
    })
  }
});

const Root = createAppContainer(MainNavigator);

const App = () => {
  const colorScheme = useColorScheme();
  const dynamicTheme = colorScheme === "dark" ? DarkTheme : DefaultTheme;
  const theme = Platform.OS === "web" ? DefaultTheme : dynamicTheme;

  return (
    <Provider theme={theme}>
      <Root />
    </Provider>
  );
};

export default App;
