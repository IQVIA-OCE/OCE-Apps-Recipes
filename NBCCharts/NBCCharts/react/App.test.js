import React from 'react';
import { NativeModules } from 'react-native';
import App from './App';
import renderer from 'react-test-renderer';
import { _TESTING_ONLY_normalize_keys } from '@react-navigation/core/lib/commonjs/routers/KeyGenerator';

describe('App', () => {
  beforeEach(() => {
    jest.useFakeTimers();
    _TESTING_ONLY_normalize_keys();
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  it('should render properly', () => {
    const tree = renderer.create(
      <App />
    );

    expect(tree).toMatchSnapshot();
  })

});
