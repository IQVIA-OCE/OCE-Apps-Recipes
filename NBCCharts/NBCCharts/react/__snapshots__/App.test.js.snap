// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render properly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={false}
    forwardedRef={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    style={
      Object {
        "backgroundColor": "#eee",
        "flex": 1,
        "flexDirection": "column-reverse",
        "overflow": "hidden",
      }
    }
  >
    <RNSScreenContainer
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        collapsable={false}
        forwardedRef={[Function]}
        nativeID="animatedComponent"
        onComponentRef={[Function]}
        pointerEvents="auto"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "opacity": 1,
            "paddingTop": 64,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          collapsable={false}
          nativeID="animatedComponent"
          style={
            Object {
              "backgroundColor": "#fff",
              "flex": 1,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "column",
                  "minHeight": "100%",
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#ffffff",
                    "flexDirection": "column",
                    "justifyContent": "space-between",
                    "padding": 4,
                    "zIndex": 1,
                  }
                }
              >
                <View
                  onLayout={[Function]}
                  style={
                    Array [
                      Object {
                        "width": 176,
                      },
                      Array [
                        Object {
                          "width": "100%",
                        },
                        undefined,
                        Object {
                          "width": "100%",
                        },
                      ],
                      Object {
                        "opacity": 1,
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "borderColor": "#D9D9D9",
                          "borderRadius": 4,
                          "borderWidth": 1,
                        },
                        Object {
                          "width": "100%",
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "height": 40,
                          "minHeight": 40,
                          "overflow": "hidden",
                          "paddingBottom": 0,
                          "paddingTop": 0,
                        }
                      }
                    >
                      <View
                        accessible={true}
                        focusable={false}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                        style={
                          Object {
                            "left": 10,
                            "opacity": 1,
                            "position": "absolute",
                            "right": "auto",
                            "top": "50%",
                            "transform": Array [
                              Object {
                                "translateY": -12.5,
                              },
                            ],
                            "zIndex": 5,
                          }
                        }
                      >
                        <Text
                          accessibilityElementsHidden={true}
                          allowFontScaling={false}
                          importantForAccessibility="no-hide-descendants"
                          pointerEvents="none"
                          style={
                            Array [
                              Object {
                                "color": "#999",
                                "fontSize": 25,
                              },
                              Array [
                                Object {
                                  "transform": Array [
                                    Object {
                                      "scaleX": 1,
                                    },
                                  ],
                                },
                                Object {
                                  "backgroundColor": "transparent",
                                },
                              ],
                              Object {
                                "fontFamily": "Material Design Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                        >
                          Ôçâ
                        </Text>
                      </View>
                      <TextInput
                        editable={true}
                        maxResizableWidth={750}
                        minHeight={60}
                        minWidth={0}
                        onBlur={[Function]}
                        onChangeText={[Function]}
                        onFocus={[Function]}
                        paddingLeft={40}
                        paddingRight={8}
                        placeholder="Search"
                        placeholderTextColor="rgba(0, 0, 0, 0.54)"
                        selectionColor="#0768FD"
                        style={
                          Array [
                            Object {
                              "margin": 0,
                              "textAlign": "left",
                              "zIndex": 5,
                            },
                            undefined,
                            Object {
                              "color": "#000000",
                              "fontFamily": "System",
                              "fontSize": 16,
                              "fontWeight": undefined,
                              "height": 40,
                              "minHeight": 0,
                              "paddingLeft": 40,
                              "paddingRight": 8,
                              "paddingVertical": 5,
                              "textAlignVertical": "center",
                              "width": "100%",
                            },
                          ]
                        }
                        value=""
                      />
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                    }
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                        "paddingHorizontal": 21,
                        "paddingVertical": 10,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#2bb3fe",
                          "fontSize": 16,
                        }
                      }
                    >
                      Sort By: 
                      Sample eligibility
                    </Text>
                  </View>
                  <View
                    collapsable={false}
                  />
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      Object {
                        "disabled": false,
                      }
                    }
                    accessible={true}
                    centered={true}
                    focusable={true}
                    hitSlop={
                      Object {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onClick={[Function]}
                    onLayout={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    pointerEvents="box-only"
                    style={Object {}}
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Array [
                            Object {
                              "alignItems": "center",
                              "justifyContent": "center",
                              "margin": 6,
                              "overflow": "hidden",
                            },
                            Object {
                              "borderRadius": 15,
                              "height": 30,
                              "width": 30,
                            },
                            undefined,
                            undefined,
                          ],
                        ]
                      }
                    >
                      <View>
                        <Text
                          accessibilityElementsHidden={true}
                          allowFontScaling={false}
                          importantForAccessibility="no-hide-descendants"
                          pointerEvents="none"
                          style={
                            Array [
                              Object {
                                "color": "#2bb3fe",
                                "fontSize": 20,
                              },
                              Array [
                                Object {
                                  "transform": Array [
                                    Object {
                                      "scaleX": 1,
                                    },
                                  ],
                                },
                                Object {
                                  "backgroundColor": "transparent",
                                },
                              ],
                              Object {
                                "fontFamily": "Material Design Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                              Object {},
                            ]
                          }
                        >
                          ÔÅÖ
                        </Text>
                      </View>
                    </View>
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "bottom": 0,
                            "left": 0,
                            "overflow": "hidden",
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          },
                          Object {
                            "borderRadius": 0,
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "backgroundColor": "#f2f2f2",
                    "flex": 1,
                    "paddingVertical": 5,
                  }
                }
              >
                <RCTScrollView
                  ListFooterComponent={[Function]}
                  data={Array []}
                  extraData={
                    Object {
                      "columns": Array [
                        Object {
                          "accessor": "OCE__Account__r.OCE__PrimaryAccountAddress__r.OCE__SampleEligibilityStatus__c",
                          "id": "SampleEligibility",
                          "isSortable": true,
                          "title": "Sample eligibility",
                        },
                        Object {
                          "accessor": "OCE__Account__r.OCE__PrimaryAccountAddress__r.OCE__BestTime__c",
                          "id": "NextBestTime",
                          "isSortable": false,
                          "title": "Next best time",
                        },
                        Object {
                          "accessor": "OCE__Rank__c",
                          "id": "Rank",
                          "isSortable": true,
                          "title": "Rank",
                          "type": "number",
                        },
                      ],
                      "dataSource": Array [],
                      "error": null,
                      "fetchDone": false,
                      "filterQuery": "",
                      "limit": 20,
                      "loading": true,
                      "offset": 0,
                      "sortColumn": Object {
                        "accessor": "OCE__Account__r.OCE__PrimaryAccountAddress__r.OCE__SampleEligibilityStatus__c",
                        "id": "SampleEligibility",
                        "isSortable": true,
                        "title": "Sample eligibility",
                      },
                      "sortDirection": "asc",
                      "sortOptionsMenuCoord": Object {
                        "x": 0,
                        "y": 0,
                      },
                      "sortOptionsVisible": false,
                      "sortingColumns": Array [
                        Object {
                          "accessor": "OCE__Account__r.OCE__PrimaryAccountAddress__r.OCE__SampleEligibilityStatus__c",
                          "id": "SampleEligibility",
                          "isSortable": true,
                          "title": "Sample eligibility",
                        },
                        Object {
                          "accessor": "OCE__Rank__c",
                          "id": "Rank",
                          "isSortable": true,
                          "title": "Rank",
                          "type": "number",
                        },
                        Object {
                          "accessor": "OCE__Account__r.LastName",
                          "id": "Last Name",
                          "title": "LastName",
                        },
                      ],
                    }
                  }
                  getItem={[Function]}
                  getItemCount={[Function]}
                  keyExtractor={[Function]}
                  onContentSizeChange={[Function]}
                  onEndReached={[Function]}
                  onEndReachedThreshold={0.5}
                  onLayout={[Function]}
                  onMomentumScrollBegin={[Function]}
                  onMomentumScrollEnd={[Function]}
                  onScroll={[Function]}
                  onScrollBeginDrag={[Function]}
                  onScrollEndDrag={[Function]}
                  removeClippedSubviews={false}
                  renderItem={[Function]}
                  scrollEventThrottle={50}
                  stickyHeaderIndices={Array []}
                  viewabilityConfigCallbackPairs={Array []}
                >
                  <View>
                    <View
                      onLayout={[Function]}
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "center",
                              "justifyContent": "center",
                            },
                            Object {
                              "minHeight": 80,
                              "paddingVertical": 10,
                            },
                          ]
                        }
                      >
                        <View
                          collapsable={false}
                          nativeID="animatedComponent"
                          style={
                            Object {
                              "height": 24,
                              "opacity": 1,
                              "width": 24,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            nativeID="animatedComponent"
                            style={
                              Object {
                                "alignItems": "center",
                                "bottom": 0,
                                "justifyContent": "center",
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "height": 24,
                                  "transform": Array [
                                    Object {
                                      "rotate": "45deg",
                                    },
                                  ],
                                  "width": 24,
                                }
                              }
                            >
                              <View
                                collapsable={false}
                                style={
                                  Object {
                                    "height": 12,
                                    "overflow": "hidden",
                                    "width": 24,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "height": 24,
                                      "transform": Array [
                                        Object {
                                          "translateY": 0,
                                        },
                                        Object {
                                          "rotate": "-165deg",
                                        },
                                      ],
                                      "width": 24,
                                    }
                                  }
                                >
                                  <View
                                    collapsable={false}
                                    style={
                                      Object {
                                        "height": 12,
                                        "overflow": "hidden",
                                        "width": 24,
                                      }
                                    }
                                  >
                                    <View
                                      collapsable={false}
                                      nativeID="animatedComponent"
                                      style={
                                        Object {
                                          "borderColor": "#0768FD",
                                          "borderRadius": 12,
                                          "borderWidth": 2.4,
                                          "height": 24,
                                          "width": 24,
                                        }
                                      }
                                    />
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                          <View
                            collapsable={false}
                            nativeID="animatedComponent"
                            style={
                              Object {
                                "alignItems": "center",
                                "bottom": 0,
                                "justifyContent": "center",
                                "left": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 0,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "height": 24,
                                  "transform": Array [
                                    Object {
                                      "rotate": "45deg",
                                    },
                                  ],
                                  "width": 24,
                                }
                              }
                            >
                              <View
                                collapsable={false}
                                style={
                                  Object {
                                    "height": 12,
                                    "overflow": "hidden",
                                    "top": 12,
                                    "width": 24,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "height": 24,
                                      "transform": Array [
                                        Object {
                                          "translateY": -12,
                                        },
                                        Object {
                                          "rotate": "345deg",
                                        },
                                      ],
                                      "width": 24,
                                    }
                                  }
                                >
                                  <View
                                    collapsable={false}
                                    style={
                                      Object {
                                        "height": 12,
                                        "overflow": "hidden",
                                        "width": 24,
                                      }
                                    }
                                  >
                                    <View
                                      collapsable={false}
                                      nativeID="animatedComponent"
                                      style={
                                        Object {
                                          "borderColor": "#0768FD",
                                          "borderRadius": 12,
                                          "borderWidth": 2.4,
                                          "height": 24,
                                          "width": 24,
                                        }
                                      }
                                    />
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </RCTScrollView>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        backTitleVisible={true}
        cardOverlayEnabled={false}
        collapsable={false}
        index={0}
        isLandscape={false}
        layout={
          Object {
            "height": 0,
            "initHeight": 0,
            "initWidth": 0,
            "isMeasured": false,
            "width": 0,
          }
        }
        layoutPreset="center"
        mode="float"
        nativeID="animatedComponent"
        navigation={
          Object {
            "actions": Object {
              "dismiss": [Function],
              "goBack": [Function],
              "navigate": [Function],
              "pop": [Function],
              "popToTop": [Function],
              "push": [Function],
              "replace": [Function],
              "reset": [Function],
              "setParams": [Function],
            },
            "addListener": [Function],
            "dangerouslyGetParent": [Function],
            "dismiss": [Function],
            "dispatch": [Function],
            "emit": [Function],
            "getChildNavigation": [Function],
            "getParam": [Function],
            "getScreenProps": [Function],
            "goBack": [Function],
            "isFirstRouteInParent": [Function],
            "isFocused": [Function],
            "navigate": [Function],
            "pop": [Function],
            "popToTop": [Function],
            "push": [Function],
            "replace": [Function],
            "reset": [Function],
            "router": undefined,
            "setParams": [Function],
            "state": Object {
              "key": "id-0",
              "routeName": "AccountsList",
            },
          }
        }
        navigationOptions={
          Object {
            "title": "Accounts",
          }
        }
        onGestureBegin={[Function]}
        onGestureCanceled={[Function]}
        onGestureEnd={[Function]}
        position={0}
        scene={
          Object {
            "descriptor": Object {
              "getComponent": [Function],
              "key": "id-0",
              "navigation": Object {
                "actions": Object {
                  "dismiss": [Function],
                  "goBack": [Function],
                  "navigate": [Function],
                  "pop": [Function],
                  "popToTop": [Function],
                  "push": [Function],
                  "replace": [Function],
                  "reset": [Function],
                  "setParams": [Function],
                },
                "addListener": [Function],
                "dangerouslyGetParent": [Function],
                "dismiss": [Function],
                "dispatch": [Function],
                "emit": [Function],
                "getChildNavigation": [Function],
                "getParam": [Function],
                "getScreenProps": [Function],
                "goBack": [Function],
                "isFirstRouteInParent": [Function],
                "isFocused": [Function],
                "navigate": [Function],
                "pop": [Function],
                "popToTop": [Function],
                "push": [Function],
                "replace": [Function],
                "reset": [Function],
                "router": undefined,
                "setParams": [Function],
                "state": Object {
                  "key": "id-0",
                  "routeName": "AccountsList",
                },
              },
              "options": Object {
                "header": [Function],
                "headerStyle": Object {
                  "backgroundColor": "transparent",
                },
                "title": "Accounts",
              },
              "state": Object {
                "key": "id-0",
                "routeName": "AccountsList",
              },
            },
            "index": 0,
            "isActive": true,
            "isStale": false,
            "key": "scene_id-0",
            "route": Object {
              "key": "id-0",
              "routeName": "AccountsList",
            },
          }
        }
        scenes={
          Array [
            Object {
              "descriptor": Object {
                "getComponent": [Function],
                "key": "id-0",
                "navigation": Object {
                  "actions": Object {
                    "dismiss": [Function],
                    "goBack": [Function],
                    "navigate": [Function],
                    "pop": [Function],
                    "popToTop": [Function],
                    "push": [Function],
                    "replace": [Function],
                    "reset": [Function],
                    "setParams": [Function],
                  },
                  "addListener": [Function],
                  "dangerouslyGetParent": [Function],
                  "dismiss": [Function],
                  "dispatch": [Function],
                  "emit": [Function],
                  "getChildNavigation": [Function],
                  "getParam": [Function],
                  "getScreenProps": [Function],
                  "goBack": [Function],
                  "isFirstRouteInParent": [Function],
                  "isFocused": [Function],
                  "navigate": [Function],
                  "pop": [Function],
                  "popToTop": [Function],
                  "push": [Function],
                  "replace": [Function],
                  "reset": [Function],
                  "router": undefined,
                  "setParams": [Function],
                  "state": Object {
                    "key": "id-0",
                    "routeName": "AccountsList",
                  },
                },
                "options": Object {
                  "header": [Function],
                  "headerStyle": Object {
                    "backgroundColor": "transparent",
                  },
                  "title": "Accounts",
                },
                "state": Object {
                  "key": "id-0",
                  "routeName": "AccountsList",
                },
              },
              "index": 0,
              "isActive": true,
              "isStale": false,
              "key": "scene_id-0",
              "route": Object {
                "key": "id-0",
                "routeName": "AccountsList",
              },
            },
          ]
        }
        shadowEnabled={true}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#0768FD",
            "elevation": 4,
            "flexDirection": "row",
            "height": 56,
            "paddingHorizontal": 4,
            "shadowColor": "#000000",
            "shadowOffset": Object {
              "height": 3,
              "width": 0,
            },
            "shadowOpacity": 0.24,
            "shadowRadius": 4,
          }
        }
        transitionPreset="fade-in-place"
      >
        <View
          style={
            Object {
              "width": 48,
            }
          }
        />
        <View
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "flex": 1,
                "paddingHorizontal": 12,
              },
              Array [
                false,
                Object {
                  "alignItems": "center",
                },
                undefined,
              ],
            ]
          }
        >
          <Text
            accessibilityRole="header"
            accessibilityTraits="header"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                  "writingDirection": "ltr",
                },
                Array [
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "400",
                  },
                  Object {
                    "fontSize": 17,
                  },
                  undefined,
                ],
              ]
            }
          >
            Accounts
          </Text>
        </View>
        <View
          style={
            Object {
              "width": 48,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
